### PR TITLE
Add PVS-Studio to our static-analysis workflow

### PR DESCRIPTION
**Pull request by krcroft**
This PR adds PVS-Studio static analysis to our existing analysis workflow. It's a stand-alone tool and generates a nice HTML report with embedded source snippets (uploaded as a zipped asset). 

It flags errors that appear to be relatively unique versus those reported by Clang's and Coverity's checkers, so it's valuable and interesting in that regard.

Output can be customized in various formats (stdout, gcc-error format, or HTML), and it produces a nice summary of results that (for a later PR) we can capture and compare against a maximum warning count similar to our other checkers.

---

From their website: PVS-Studio is a tool for detecting bugs and security weaknesses in the
source code of programs, written in C, C++, C# and Java. It works under
64-bit systems in Windows, Linux and macOS environments, and can analyze
source code intended for 32-bit, 64-bit and embedded ARM platforms.

https://www.viva64.com/en/pvs-studio/

### CODE REVIEW

File: .github/workflows/analysis.yml [hunk 1]
```diff
@@ -66,6 +66,39 @@ jobs:
...
+      - uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
+      - run:  sudo apt-get update
+      - name: Log environment
+        run:  ./scripts/log-env.sh
+      - name: Install packages
+        run:  |
+          set -xeu
+          wget -nv "https://files.viva64.com/pvs-studio-7.05.35582.25-amd64.deb" -O "pvs.deb" &
```
Comment (dreamer) [inline .github/workflows/analysis.yml]: This step seems like a perfect candidate for using cache.

Comment (dreamer) [reply .github/workflows/analysis.yml]: Argh; I meant specifically line 83.

Comment (krcroft) [reply .github/workflows/analysis.yml]: Great suggestion; will give it a try!

**Comment by dreamer** [comment]
Great, thanks! :)

Let's see what new tool can detect…

**Pull request merged**